pkg/resource/table/hooks_continuous_backup.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package table
+
+import (
+	"context"
+
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/dynamodb"
+
+	"github.com/aws-controllers-k8s/dynamodb-controller/apis/v1alpha1"
+)
+
+// syncContinuousBackup syncs the PointInTimeRecoverySpecification of the dynamodb table.
+func (rm *resourceManager) syncContinuousBackup(
+	ctx context.Context,
+	desired *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncContinuousBackup")
+	defer func(err error) { exit(err) }(err)
+
+	pitrSpec := &svcsdk.PointInTimeRecoverySpecification{}
+	if desired.ko.Spec.ContinuousBackups != nil &&
+		desired.ko.Spec.ContinuousBackups.PointInTimeRecoveryEnabled != nil {
+		pitrSpec.SetPointInTimeRecoveryEnabled(
+			*desired.ko.Spec.ContinuousBackups.PointInTimeRecoveryEnabled,
+		)
+	}
+
+	_, err = rm.sdkapi.UpdateContinuousBackupsWithContext(
+		ctx,
+		&svcsdk.UpdateContinuousBackupsInput{
+			TableName:                        desired.ko.Spec.TableName,
+			PointInTimeRecoverySpecification: pitrSpec,
+		},
+	)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateContinuousBackups", err)
+	return err
+}
+
+// getResourcePointInTimeRecoveryWithContext gets the PointInTimeRecoverySpecification of the dynamodb table.
+func (rm *resourceManager) getResourcePointInTimeRecoveryWithContext(
+	ctx context.Context,
+	tableName *string,
+) (*v1alpha1.PointInTimeRecoverySpecification, error) {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.getResourcePointInTimeRecoveryWithContext")
+	defer func(err error) { exit(err) }(err)
+
+	res, err := rm.sdkapi.DescribeContinuousBackupsWithContext(
+		ctx,
+		&svcsdk.DescribeContinuousBackupsInput{
+			TableName: tableName,
+		},
+	)
+
+	rm.metrics.RecordAPICall("GET", "DescribeContinuousBackups", err)
+	if err != nil {
+		return nil, err
+	}
+
+	isEnabled := false
+	if res.ContinuousBackupsDescription != nil {
+		isEnabled = *res.ContinuousBackupsDescription.PointInTimeRecoveryDescription.PointInTimeRecoveryStatus == svcsdk.PointInTimeRecoveryStatusEnabled
+	}
+
+	return &v1alpha1.PointInTimeRecoverySpecification{
+		PointInTimeRecoveryEnabled: &isEnabled,
+	}, nil
```

Comment:
This was the case for the first version I pushed. I since then fixed the issue and force-pushed to my branch. For me its no longer infinitely updating. I just get the log once and then the pitr is updated on the table

Reply:
Was the table synced? (status.Conditions[ACKSynced = true])

Reply:
For me yes. This is the only log message I get when for example setting pitr from `enabled` to `disabled`:

```
2023-07-19T22:28:56.163Z        INFO    ackrt   desired resource state has changed      {"account": "223727272", "role": "", "region": "eu-central-1", "kind": "Table", "namespace": "default", "name": "my-dynamodb", "is_adopted": false, "generation": 27, "diff": [{"Path":{"Parts":["Spec","ContinuousBackups","PointInTimeRecoveryEnabled"]},"A":false,"B":true}]}
```

It is only displayed once. Table status shows:

```
Status:
  Ack Resource Metadata:
    Arn:               arn:aws:dynamodb:eu-central-1:404164987772:table/tenant-config
    Owner Account ID:  404164987772
    Region:            eu-central-1
  Conditions:
    Last Transition Time:  2023-07-19T22:29:01Z
    Message:               Resource synced successfully
    Reason:
    Status:                True
    Type:                  ACK.ResourceSynced
```